src/modules/PrototypeConnection.py

```diff
@@ -59,29 +60,48 @@ def connect_with_config(self, fp_json: str):
         # set found device if not in backend-debug mode
         if not self.debug:
```

Comment:
same as my comment below: if you invert this `if` statement, you can reduce the number of indents you need.
 ```suggestion
        if self.debug:
            Logger.log_info("PrototypeConnection initialized in debug mode.")
            return
        
        try:
            port = None 
        ...
```